### DISCUSSION TITLE
Separating applications from the module

### DISCUSSION THREAD

**Discussion by aaronparker** [EUCPilots/evergreen-module#236] [Ideas]
I've been thinking about how to seperate the applications (manifests and internal functions) from the module so that an update to an application isn't tied to an update to the module. 

At this point, I have an idea that I'd write `Update-EvergreenApp` or similar which would download the internal functions and manifests from the repo into a local directory (e.g., `C:\ProgramData\Evergreen`). 

Can anyone see an issue with this approach or think of a better solution?

**Comment by msfreaks** [comment]
Maybe check for updated manifests / functions when the module is loaded? I have done this in the .psm1 file for instance.

Many of us use the module in automated processes which means the module gets loaded each run, which for me would be a perfect time to check for updates.

If you're introducing the `Update-EvergreenApp` or simular function this would mean I have to update everything to first call that update before using `Get-EvergreenApp`.
Unless `Update-EvergreenApp` would update all manifests and functions, that would work for me I guess.

If you do decide to go this way, please let the function decide if it's running the module elevated or not. If elevated `C:\ProgramData\Evergreen`, if not put it in the current user's appdata folder or something.

**Reply by DanGough** [reply]
Agree with the elevated thing above. ProgramData permissions are a little odd - standard users can create new folders/files there, but can't write to a folder that was created by someone else since they aren't the owner of it.
